packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx

```diff
@@ -1,40 +1,83 @@
-import {Component, h, Host, Listen, State} from '@stencil/core';
+import {Component, h, Host, State, Element, Method} from '@stencil/core';
 
-export interface UpdateLiveRegionEventArgs {
-  region: string;
-  message: string;
+export interface FindAriaLiveEventArgs {
+  element?: HTMLAtomicAriaLiveElement;
 }
 
 /**
  * The `atomic-aria-live` component notifies screen readers of changes in the search interface.
- * @internal
+ *
+ * We do not recommend dynamically adding/removing this component from the page. It is better to have this component during the initial render of the page.
  */
 @Component({
   tag: 'atomic-aria-live',
   shadow: false,
 })
 export class AtomicAriaLive {
-  @State() public regions: Record<string, string> = {};
+  @Element() private host!: HTMLAtomicAriaLiveElement;
+  @State() private message = '';
 
-  @Listen('atomic/accessibility/updateLiveRegion', {target: 'document'})
-  protected updateMessage({
-    detail: {message, region},
-  }: CustomEvent<UpdateLiveRegionEventArgs>) {
-    this.regions = {...this.regions, [region]: message};
+  private lastUpdatedRegion?: string;
+  private disconnectFindAriaLiveEvent?: () => void;
+
+  protected onFindAriaLive(args: FindAriaLiveEventArgs) {
+    if (!args.element || !this.isInSearchInterface) {
+      args.element = this.host;
+    }
+  }
+
+  private get isInSearchInterface() {
+    let element: Element | null = this.host;
+    while (element) {
+      if (element.tagName === 'ATOMIC-SEARCH-INTERFACE') {
+        return true;
+      }
+      element = element.parentElement;
+    }
+    return false;
+  }
+
+  /**
+   * @internal
+   */
+  @Method()
+  public async updateMessage(region: string, message: string) {
+    const wouldOverwriteAnotherRegion = region !== this.lastUpdatedRegion;
+    if (wouldOverwriteAnotherRegion) {
+      this.lastUpdatedRegion = region;
+      if (!message) {
+        return;
+      }
+    }
+    this.message = message;
+  }
+
+  public connectedCallback() {
+    const eventListener = (ev: Event) =>
+      this.onFindAriaLive((ev as CustomEvent<FindAriaLiveEventArgs>).detail);
+    document.addEventListener(
+      'atomic/accessibility/findAriaLive',
```

Comment:
now that your not using a decorator, you can put the event value inside a variable & reuse it directly